automation/features/step-definitions/staking.steps.js

```diff
@@ -1,70 +1,349 @@
+      });
+      stakingPage.removeStakeBtn.click({ force: true });
+      break;
+    default:
+      throw new Error(`${stakeAction} stake Not an option`);
+  }
+});
+
+Then(/^I can submit the stake successfully$/, () => {
+  expect(stakingPage.tokenAmountSubmitBtn).toBeEnabled();
+  stakingPage.tokenAmountSubmitBtn.click();
+});
+
+When(
+  /^I can see "([^"]*)?" vega has been removed from staking$/,
+  (tokenAmount) => {
+    stakingPage.calloutSuccess.waitForDisplayed({
+      timeout: 60000,
+      timeoutMsg:
+        "remove stake success message container not displayed after 60 seconds",
+    });
+    expect(
+      $(`.callout__title*=${tokenAmount} $VEGA has been removed from validator`)
+    ).toBeDisplayed();
+    expect(
+      stakingPage.getBypTag("It will be applied in the next epoch")
+    ).toBeDisplayed();
+  }
+);
+
+When(/^I click on the option to remove stake now$/, () => {
+  stakingPage.removeStakeNowBtn.waitForDisplayed({
+    timeout: 20000,
+    timeoutMsg: "remove stake now button not displayed",
+  });
+  stakingPage.removeStakeNowBtn.click();
+});
+
+Then(/^I can submit successfully$/, () => {
+  expect(stakingPage.tokenAmountSubmitBtn).toBeEnabled();
+  stakingPage.tokenAmountSubmitBtn.click();
+});
+
+Then(/^the stake is successful$/, () => {
+  stakingPage.calloutSuccess.waitForDisplayed({
+    timeout: 60000,
+    timeoutMsg:
+      "stake success message container not displayed after 60 seconds",
+  });
+  expect(
+    stakingPage.getBypTag(
+      ".callout__title=At the beginning of the next epoch your $VEGA will be nominated to the validator"
+    )
+  ).toBeDisplayed();
+});
+
+Then(/^I am back on the staking main page$/, () => {
+  expect(browser.getUrl()).toEqual(`${browser.options.baseUrl}/staking`);
+});
+
+When(/^I click the pending transactions button$/, () => {
+  stakingPage.pendingTransactionsBtn.click();
+});
+
+Then(/^I can see the pending transactions modal is shown$/, () => {
+  stakingPage.pendingTransactionsModal.waitForDisplayed({
+    timeout: 20000,
+    timeoutMsg: "pending transactions modal did not display",
+  });
+  expect(stakingPage.etherscanLink).toBeDisplayed();
+  expect(stakingPage.etherscanLink).toHaveAttr("href");
+  expect(stakingPage.pendingTransactionsModalStatus).toHaveText("Pending");
+});
+
+Then(/^the pending transactions modal can be closed$/, () => {
+  $("main").click({ force: true });
+  expect(stakingPage.pendingTransactionsModal).not.toBeDisplayed();
+});
+
+When(/^the association of "([^"]*)?" has been successful$/, (tokenAmount) => {
+  expect(
+    browser.waitUntil(
+      () =>
+        stakingPage.associatedinWalletAmountText.getText() !==
+        tokensAssociatedInVegaWalletText,
+      {
+        timeout: 60000,
+        timeoutMsg: "expected balance to be different",
+      }
+    )
+  ).toBeTruthy();
+});
+
+When(
+  /^the disassociation of "([^"]*)?" has been successful$/,
+  (tokenAmount) => {
```

Comment:
question (blocking): don't we need to do some maths here with the tokenAmount and check balances? Same for linting error above?

Reply:
Or a TODO comment to add this when we can do this determinisitically?